lib/wallaroo/core/sink/connector_sink/connector_sink.pony

```diff
@@ -303,16 +303,10 @@ actor ConnectorSink is Sink
       // We do not include MessageMsg size overhead in our offset accounting
       _twopc.update_offset(encoded1_len)
 
-      let encoded2 =
-        try
-          let w1: Writer = w1.create()
-          let msg = _notify.make_message(encoded1)?
-          let bs = cwm.Frame.encode(msg, w1)
-          Bytes.length_encode(bs)
-        else
-          Fail()
-          encoded1
-        end
+      let w1: Writer = w1.create()
```

Comment:
If this is not a one time line, then we should be reusing a `Writer` that we maintain in a field.  We just need to make sure `.done()` is called at the correct time.

Reply:
@slfritchie I'll defer this one to you if that's ok

Reply:
I think we can do the Reader & Writer sharing in another PR.